### PR TITLE
fix: decoding nested dicts with tuple keys

### PR DESCRIPTION
**Pull request by pearceeverydaylabs**
suggested fix for https://github.com/lidatong/dataclasses-json/issues/294

### CODE REVIEW

**Review by christian-steinmeyer — COMMENTED** [review]
Should do the trick!

**Comment by pearceeverydaylabs** [comment]
@lidatong could you review, please?

File: dataclasses_json/core.py [hunk 1]
```diff
@@ -285,11 +285,20 @@ def _decode_dict_keys(key_type, xs, infer_missing):
...
+    if key_type is None or key_type == Any:
+        decode_function = key_type = (lambda x: x)
+    # handle a nested python dict that has tuples for keys. E.g. for
+    # Dict[Tuple[int], int], key_type would be typing.Tuple[int], but
+    # decode_function would be tuple, so map() doesn't break
+    # the key_type would be typing.Tuple[int]
+    elif _get_type_origin(key_type) == tuple:
+        decode_function = tuple
+        key_type = key_type
+
```
Comment (lidatong) [inline dataclasses_json/core.py:300]: what would happen in the `else` case?

Comment (lidatong) [reply dataclasses_json/core.py:300]: never mind i see the original logic was effectively `key_type` = `key_type`

**Review by lidatong — APPROVED** [review]
thanks for the contribution! appreciate it

**Comment by pearceeverydaylabs** [comment]
@lidatong thanks for approving! I pushed up a fix for python 3.6 that I got working locally in a 3.6 venv. Could you please approve re-running workflows since I don't have permission? :) 
Screen Shot 2021-06-04 at 12 20 10 AM

**Comment by lidatong** [comment]
Sure -- and looks like checks are passing now. Thanks again for the contribution!

**Pull request merged**